Sources/HAP/Endpoints/pairVerify().swift

```diff
@@ -63,7 +63,45 @@ func pairVerify(device: Device) -> Responder {
         } catch {
             logger.warning("Could not verify pairing: \(error)")
             setSession(for: context, to: nil)
-            return .badRequest
+
+            // HAP Specification 4.8.4:
+            // Must return authentication error to Verify Finish Response (M4) when
+            // 1. fail verifiction of authentication tag on encrypted keys
+            // 2. decryption fails
+            // 3. pairing ID is not recognised
+            // 4. fail verification of iOS device signature
+
+            let response: PairTagTLV8?
+
+            switch error {
```

Comment:
Should we only return an authentication error for this limited set of errors, or could we do it always? If we could, it would simplify things here.

Reply:
To correctly implement the spec requirements, this error should only be raised for these specific errors.